### ISSUE TITLE
All bars have disappeared after today's WoTLK update

### ISSUE THREAD

**Issue by DenneyLewis** [Nevcairiel/Bartender4#78]
As title suggest, the bars are all gone. No workaround found so far.

**Comment by jmhickman** [comment]
Just came here to post this.

**Comment by nivalen292** [comment]
Same

**Comment by DenneyLewis** [comment]
Message: Interface\AddOns\Bartender4\Options\ActionBar.lua:34: Invalid bar id in options table. (1)
Time: Fri Sep  2 14:22:20 2022
Count: 1
Stack: Interface\AddOns\Bartender4\Options\ActionBar.lua:34: Invalid bar id in options table. (1)
[string "=[C]"]: ?
[string "@Interface\AddOns\Bartender4\Options\ActionBar.lua"]:34: in function <Interface\AddOns\Bartender4\Options\ActionBar.lua:32>
[string "@Interface\AddOns\Bartender4\Options\ActionBar.lua"]:47: in function `member'
[string "@Interface\AddOns\Ace3\AceConfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua"]:221: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:169>
[string "@Interface\AddOns\Ace3\AceConfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua"]:1206: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:1121>
[string "@Interface\AddOns\Ace3\AceConfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua"]:1652: in function `FeedGroup'
[string "@Interface\AddOns\Ace3\AceConfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua"]:1577: in function <...nfig-3.0\AceConfigDialog-3.0\AceConfigDialog-3.0.lua:1561>
[string "=[C]"]: ?
[string "@Interface\AddOns\Ace3\AceGUI-3.0\AceGUI-3.0.lua"]:72: in function <Interface\AddOns\Ace3\AceGUI-3.0\AceGUI-3.0.lua:70>
[string "@Interface\AddOns\Ace3\AceGUI-3.0\AceGUI-3.0.lua"]:306: in function `Fire'
[string "@Interface\AddOns\Ace3\AceGUI-3.0\widgets\AceGUIContainer-TabGroup.lua"]:156: in function `SelectTab'
[string "@Interface\AddOns\Ace3\AceGUI-3.0\widgets\AceGUIContainer-TabGroup.lua"]:67: in function <...Ace3\AceGUI-3.0\widgets\AceGUIContainer-TabGroup.lua:64>

Locals: <none>

In case this helps.

**Comment by Bosenator** [comment]
```
1x ...er4\libs\LibActionButton-1.0\LibActionButton-1.0-83.lua:683: Attempt to register unknown event "ARCHAEOLOGY_CLOSED"
[string "=[C]"]: in function `RegisterEvent'
[string "@Bartender4\libs\LibActionButton-1.0\LibActionButton-1.0-83.lua"]:683: in function <...er4\libs\LibActionButton-1.0\LibActionButton-1.0.lua:650>
[string "@Bartender4\libs\LibActionButton-1.0\LibActionButton-1.0-83.lua"]:183: in function `CreateButton'
[string "@Bartender4\ActionBar.lua"]:217: in function `UpdateButtons'
[string "@Bartender4\ActionBar.lua"]:37: in function `ApplyConfig'
[string "@Bartender4\ActionBars.lua"]:191: in function `Create'
[string "@Bartender4\ActionBars.lua"]:81: in function <Bartender4\ActionBars.lua:73>
[string "=[C]"]: ?
[string "@Attune\Libs\AceAddon-3.0\AceAddon-3.0-13.lua"]:70: in function <...aceAttune\Libs\AceAddon-3.0\AceAddon-3.0.lua:65>
[string "@Attune\Libs\AceAddon-3.0\AceAddon-3.0-13.lua"]:527: in function `EnableAddon'
[string "@Attune\Libs\AceAddon-3.0\AceAddon-3.0-13.lua"]:540: in function `EnableAddon'
[string "@Attune\Libs\AceAddon-3.0\AceAddon-3.0-13.lua"]:630: in function <...aceAttune\Libs\AceAddon-3.0\AceAddon-3.0.lua:615>```

Looks like the conditional on it checking if it's WoTLK has broken with the WoW Client update.

**Comment by sintaxx** [comment]
Did someone not test this build before pushing it? It was working 100% fine until this update.

**Comment by Wutname1** [comment]
> Did someone not test this build before pushing it? It was working 100% fine until this update.

Has nothing to do with BT4, blizzard changed how the game tells addons what version of the game is running randomly in a hotfix. something we have been saying needs to be updated since the start of wrath beta. 

> 1x ...er4\libs\LibActionButton-1.0\LibActionButton-1.0-83.lua:683: Attempt to register unknown event "ARCHAEOLOGY_CLOSED"

This is the bug, a temporary fix is to open the file `libs\LibActionButton-1.0\LibActionButton-1.0.lua` and change `WOW_PROJECT_BURNING_CRUSADE_CLASSIC` to `WOW_PROJECT_WRATH_CLASSIC`

**Comment by DenneyLewis** [comment]
Blizzard update broke LibActionButton-1.0 (one of the libraries this AddOn uses). It wrongly assumes we are on Retail.

You can patch it yourself right now by going to Interface\AddOns\Bartender4\libs\LibActionButton-1.0 and opening the LibActionButton-1.0.lua with any text editor, find Line 43 or Line 44 and change the resutl to true.

Original: local WoWClassic = (WOW_PROJECT_ID == WOW_PROJECT_CLASSIC) local WoWBCC = true--(WOW_PROJECT_ID == WOW_PROJECT_BURNING_CRUSADE_CLASSIC)

Patched: local WoWClassic = true local WoWBCC = true

From Reddit. Not Bartenders fault. Blizzard decided to finally change something that needed to be fixed a while ago.

**Comment by Jakobud** [comment]
While your code fix does fix the addon from throwing errors, it's a bandaid and doesn't actually fix the source of the problem. It really isn't the proper way to fix it.

The better solution is to add:

```
local WoWWrath = (WOW_PROJECT_ID == WOW_PROJECT_WRATH_CLASSIC)
```

And then anywhere you see this:

```
if not WoWClassic and not WoWBCC then
  -- some code here
end
```

and change it to this:

```
if not WoWClassic and not WoWBCC and not WoWWrath then
  -- some code here
end
```

**Comment by Nevcairiel** [comment]
This issue has already been resolved in 4.11.5. Thank Blizzard for pushing a breaking update on Saturday evening.
